backend-app/routes/users/user_route.js

```diff
@@ -3,6 +3,7 @@ const swaggergenerator = require('../../utils/swagger/swaggergenerator');
 const userController = require('../../controllers/user_controller');
 const router = express.Router();
 
+router.delete('/logout', userController.logout);
```

Comment:
The auth route is not protected. So i can't get the **req.user** obj from the auth route. I need the **req.user** obj to logout the user. There is an other way where i have to validate the jwt and get the jwt payload in the auth controller. But i am not sure if this is a best practice

Reply:
Then how i will identify that the user who is trying to logout is the real user and not a hacker? If the user is a real user he have to pass the access token if the token is valid he can logout. If the access token is expired he have to refresh the token and then again pass the access token.

Reply:
well how do we get the req.user from the first place huh? 
=> we get it from decoding  access token right ? 
so why validating the token if it's in db it's gotta be correct 
@muttaqin1 
and i will open an other issue for the access token being stored in database because i think it's not a best practice especially if the app has so many logins and logouts a day

Reply:
if you want to keep it it's fine, i just think that logout belongs to the auth route 😁

Reply:
> well how do we get the req.user from the first place huh? 
> => we get it from decoding  access token right ? 
> so why validating the token if it's in db it's gotta be correct 
> @muttaqin1 
> and i will open an other issue for the access token being stored in database because i think it's not a best practice especially if the app has so many logins and logouts a day 

You have a strong logic my friend😅.

Btw if we dont create a connection with the token and database how we will logout a user. the access token comes from the authorization header not from the cookie so that we can remove it from the server side.

Reply:
> if you want to keep it it's fine, i just think that logout belongs to the auth route 😁

Sure i will move it to the auth route.

Reply:
> You have a strong logic my friend😅.

😂 no i just got lucky to think about it 

> Btw if we dont create a connection with the token and database how we will logout a user. the access token comes from the authorization header not from the cookie so that we can remove it from the server side

i didn't understand you, i think we logout the user in backend by deleting the token model from db, and refresh token and in the frontend by deleting the access token
@muttaqin1

Reply:
> > You have a strong logic my friend😅.
> 
> 😂 no i just got lucky to think about it

If the token is expired? Unlimited logout 😂
 
> 
> > Btw if we dont create a connection with the token and database how we will logout a user. the access token comes from the authorization header not from the cookie so that we can remove it from the server side
> 
> i didn't understand you, i think we logout the user in backend by deleting the token model from db, and refresh token and in the frontend by deleting the access token
> @muttaqin1 

We logout the user by deleting the token record.
We refresh the token by deleting the existing token record and create a new one.
If there is no token model how we will do  this?

Reply:
> well how do we get the req.user from the first place huh? 
> => we get it from decoding  access token right ? 
> so why validating the token if it's in db it's gotta be correct 
> @muttaqin1 
> and i will open an other issue for the access token being stored in database because i think it's not a best practice especially if the app has so many logins and logouts a day 

Do you remember the "AuthTokenKey"?
We can use that to create connection with the token and db

Reply:
```txt
log out -> delete token record (refresh token ) and deleted access token from front-end 
```
**Here we are deleting the token record from db and access token from the frontend.
Dont we have to delete the refresh token from the frontend?
**
```txt
refresh logging -> generate a new access token from refresh token that is in the db and also in httponly cookie
```
👌



The token record will only store refresh token right?

So when a user will login we will set 2 secured 🍪(Access & refresh token) in the frontend. Store refresh token in the db(Token record).
Whenever a user tries to logout we will remove the token record from db and access & refresh token from the cookie.
When the user tries to refresh token we will take the refresh token and validate it then set a new access token cookie in the fronted.

since we are using 🍪 we dont need to store the refresh token in db. we can directly remove both access and refresh token cookie to logout user. 

what do you think about this?

Reply:
That's exactly what I was thinking, just add this part :
We don't need to store any token in db since there are other ways to keep track of user activites more efficient than storing it in db
@muttaqin1